packages/slice-machine/src/features/navigation/Navigation.tsx

```diff
@@ -0,0 +1,151 @@
+import { CUSTOM_TYPES_CONFIG } from "@/features/customTypes/customTypesConfig";
+import { CUSTOM_TYPES_MESSAGES } from "@/features/customTypes/customTypesMessages";
+import { MasterSliceLibraryPreviewModal } from "@/features/masterSliceLibrary/SliceLibraryPreviewModal";
+import { RepositoryInfo } from "@/features/navigation/RepositoryInfo";
+import { OnboardingGuide } from "@/features/onboarding";
+import { useGitIntegrationExperiment } from "@/features/settings/git/useGitIntegrationExperiment";
+import { useMarketingContent } from "@/hooks/useMarketingContent";
+import { FolderIcon } from "@/icons/FolderIcon";
+import { LightningIcon } from "@/icons/Lightning";
+import { MasterSliceLibraryIcon } from "@/icons/MasterSliceLibraryIcon";
+import { SettingsIcon } from "@/icons/SettingsIcon";
+
+import { ChangesItem } from "../../legacy/components/Navigation/ChangesItem";
+import { Environment } from "../../legacy/components/Navigation/Environment";
+import { NavigationItem } from "./NavigationItem";
+import { SliceMachineVersion } from "./SliceMachineVersion";
+import { UpdateInfo } from "./UpdateInfo";
+
+export function Navigation() {
+  const router = useRouter();
+
+  const gitIntegrationExperiment = useGitIntegrationExperiment();
+  const [isSliceLibraryDialogOpen, setIsSliceLibraryDialogOpen] =
+    useState(false);
+  const { masterSliceLibrary } = useMarketingContent();
+
+  return (
+    <Box
+      as="nav"
+      flexDirection="column"
+      padding={{ top: 16, right: 16, left: 16, bottom: 0 }}
+      gap={16}
+      minWidth={0}
+    >
+      <ErrorBoundary>
+        <Suspense>
+          <Environment />
+        </Suspense>
+      </ErrorBoundary>
+
+      <Separator style="dashed" />
+
+      <RepositoryInfo />
+
+      <ChangesItem />
+
+      <Box flexDirection="column" padding={{ top: 8 }} flexGrow={1}>
+        <ActionList variant="compact">
+          <NavigationItem
+            title={CUSTOM_TYPES_MESSAGES["page"].name({
+              start: true,
+              plural: true,
+            })}
+            href={CUSTOM_TYPES_CONFIG["page"].tablePagePathname}
+            active={CUSTOM_TYPES_CONFIG["page"].matchesTablePagePathname(
+              router.asPath,
+            )}
+            Icon={CUSTOM_TYPES_CONFIG.page.Icon}
+          />
+
+          <NavigationItem
+            title={CUSTOM_TYPES_MESSAGES["custom"].name({
+              start: true,
+              plural: true,
+            })}
+            href={CUSTOM_TYPES_CONFIG["custom"].tablePagePathname}
+            active={CUSTOM_TYPES_CONFIG["custom"].matchesTablePagePathname(
+              router.asPath,
+            )}
+            Icon={CUSTOM_TYPES_CONFIG.custom.Icon}
+          />
+
+          <NavigationItem
+            title="Slices"
+            href="/slices"
+            Icon={FolderIcon}
+            active={router.asPath.startsWith("/slices")}
+          />
+        </ActionList>
+
+        <ErrorBoundary>
+          <Suspense>
```

Comment:
Here the problem is that fallback might not make much sense since we display this element only when update is available, so not that often - not sure if it's good idea to delay loading of whole navbar until we have information should we display that information or not. Will double check that with Come